### ISSUE TITLE
Add "has_notices" as plugin paramater

### ISSUE THREAD

**Issue by scrobbleme** [thomasgriffin/TGM-Plugin-Activation#648]
It would be great if plugin parameters could have a has_notices (default: true) as well.

true: current behaviour

false: 
- If "required == true" -> ignore "false"
- don't show an admin notice for this plugin

Background

We are working on a theme, where we reference some dependency which are neither required not recommended. They are just optional. Means: we just provide better support for these plugins or we generally recommend these plugins for their specific purpose. But we don't want to bother the user with the admin notices.

If there is generall interest in such a feature I may also create a pull request for this.

**Comment by jrfnl** [comment]
Interesting idea. Let's see if there are more people interested in this though before putting in time & effort.

> some dependency which are neither required not recommended.

In that case, it's not a **_dependency_** and not really what TGMPA is intended to be used for....

**Comment by scrobbleme** [comment]
> In that case, it's not a dependency and not really what TGMPA is intended to be used for....

You are right, but eventually this is correct for "recommended" dependencies as well, as the referencing plugin/theme not really depends on it ;) Still optional, just recommended with a message.
